### PR TITLE
feat: add timeout to user interactions per tick

### PR DESCRIPTION
**Pull request by guidota**
## Description

Adds the concept of tick duration. On every step, on every tick, users act sending messages. With these changes we ensure those interactions happen during the tick, otherwise they will be cancelled.

Also it adds a metric of total messages sent to the report.

### Extra
- reverted the exception when there is no room joined as it often happens (to be reviewed)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -30,7 +31,8 @@ pub struct Configuration {
     total_steps: usize,
     users_per_step: usize,
     friendship_ratio: f32,
-    step_duration_in_secs: usize,
+    step_duration_in_secs: u64,
+    tick_duration_in_secs: u64,
```
Comment (moliva) [inline src/lib.rs:35]: QQ: Can't these be `Duration`s instead? I think this was possible.

File: src/user.rs [hunk 2]
```diff
@@ -306,24 +306,23 @@ impl User<Synching> {
...
-        match response {
-            Ok(response) => {
-                self.send(Event::RequestDuration((
-                    UserRequest::SendMessage,
-                    instant.elapsed(),
-                )))
-                .await;
-
-                self.send(Event::MessageSent(response.event_id.to_string()))
+        if let Some(room) = client.get_joined_room(room_id) {
```
Comment (moliva) [inline src/user.rs:309]: Let's add a `TODO` here for the future since we need to review the issue of having users not joined in the rooms.

**Review by moliva — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (guidota) [reply src/lib.rs:35]: will tackle this on next PR 👍

File: src/user.rs (hunk 2 above)
Comment (guidota) [reply src/user.rs:309]: added on next incoming PR 👍

**Pull request merged**